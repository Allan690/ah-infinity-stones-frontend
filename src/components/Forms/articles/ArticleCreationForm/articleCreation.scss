$font-stack:    "Segoe UI", Arial, sans-serif;
$primary-color: #86c232;
$secondary-color: #222629;

// $npm run build-css will print in generated css
#articleForm {
    width:95%;
    border: 1px solid black;
    align-content: center;
    margin:2em;
    display: flex;
    padding-right: 4em;
    padding-left: 4em;
    padding-top: 2em;
    padding-bottom: 2em;
}
.article-creation-form > li {
    list-style:none;
  }
ul > li > input, textarea {
    width: 100%;
}
#articleDescription {
    height: 4em;
}
#articleBody {
    display: none;
}
#publishBtn {
    background-color: $primary-color;
    border-color: $primary-color;
    color: white;
}
#publishBtn:hover {
  color: black;
  i.icon-publish {
    color:black !important;
  }
}

label {
    font-weight: bold;
}
// editorToolbar css 
.demo-wrapper {
    border-style: solid;
    border-width: 0.5px;
    border-color: LightGray;
    span{
      color: black;
      text-align: center;
    }
}
.demo-editor {
    padding-top: 0px;
    border-style: solid;
    border-width: 0.5px;
    border-color: inherit;
    margin-left: 0.3em;
    margin-right: 0.3em;
    margin-bottom: 0.3em;
    padding-left: 0.3em;
    padding-right: 0.3em;
}
.demo-wrapper form span {
  color:black;
}

.toolbar-class {
    margin: 0.3em;
    color:black !important;
}


