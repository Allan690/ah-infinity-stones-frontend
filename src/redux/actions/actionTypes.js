export const REQUEST_LOGIN = 'REQUEST_LOGIN';
export const RECEIVE_LOGIN = 'RECEIVE_LOGIN';
export const LOGIN_ERROR = 'LOGIN_ERROR';
export const SIGNUP_REQUEST = 'SIGNUP_REQUEST';
export const SIGNUP_SUCCESS = 'SIGNUP_SUCCESS';
export const SIGNUP_FAILURE = 'SIGNUP_FAILURE';

export const SOCIAL_LOGIN_REQUEST = 'SOCIAL_LOGIN_REQUEST';
export const SOCIAL_LOGIN_SUCCESS = 'SOCIAL_LOGIN_SUCCESS';
export const SOCIAL_LOGIN_ERROR = 'SOCIAL_LOGIN_ERROR';


export const REQUEST_INCREMENT = 'REQUEST_INCREMENT';
export const RECEIVE_INCREMENT = 'RECEIVE_INCREMENT';

export const PASSWORD_RESET_START = 'PASSWORD_RESET_START';
export const PASSWORD_RESET_COMPLETE = 'PASSWORD_RESET_COMPLETE';
export const PASSWORD_RESET_ERROR = 'PASSWORD_RESET_ERROR';


export const PASSWORD_RESET_UPDATE_START = 'PASSWORD_RESET_UPDATE_START';
export const PASSWORD_RESET_UPDATE_COMPLETE = 'PASSWORD_RESET_UPDATE_COMPLETE';
export const PASSWORD_RESET_UPDATE_ERROR = 'ASSWORD_RESET_UPDATE_ERROR';
