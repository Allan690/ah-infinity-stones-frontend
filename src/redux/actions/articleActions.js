import {
  CREATE_ARTICLE, CREATE_ARTICLE_SUCCESS, CREATE_ARTICLE_FAILURE,
  GET_ALL_ARTICLES_REQUEST, GET_ALL_ARTICLES_SUCCESS, GET_ALL_ARTICLES_FAILURE,
  GET_ONE_ARTICLE_REQUEST, GET_ONE_ARTICLE_SUCCESS, GET_ONE_ARTICLE_FAILURE,
  EDIT_ARTICLE, UPDATE_ARTICLE, DELETE_ARTICLE,
} from './actionTypes';


export const createArticleAction = article => ({
  type: CREATE_ARTICLE,
  payload: article,
});
export const createArticleSuccessAction = article => ({
  type: CREATE_ARTICLE_SUCCESS,
  payload: article,
});
export const createArticleFailureAction = error => ({
  type: CREATE_ARTICLE_FAILURE,
  payload: error,
});

export const getAllArticlesRequestAction = () => ({
  type: GET_ALL_ARTICLES_REQUEST,
});

export const getAllArticlesSuccessAction = articles => ({
  type: GET_ALL_ARTICLES_SUCCESS,
  payload: [articles],
});

export const getAllArticlesFailureAction = error => ({
  type: GET_ALL_ARTICLES_FAILURE,
  payload: error,
});

export const getOneArticleRequestAction = artSlug => ({
  type: GET_ONE_ARTICLE_REQUEST,
  payload: artSlug,
});

export const getOneArticleSuccessAction = article => ({
  type: GET_ONE_ARTICLE_SUCCESS,
  payload: article,
});

export const getOneArticleFailureAction = error => ({
  type: GET_ONE_ARTICLE_FAILURE,
  payload: error,
});

export const editArticleAction = article => ({
  type: EDIT_ARTICLE,
  payload: article,
});

export const updateArticleAction = article => ({
  type: UPDATE_ARTICLE,
  payload: article,
});

export const deleteArticleAction = art_slug => ({
  type: DELETE_ARTICLE,
  payload: art_slug,
});
