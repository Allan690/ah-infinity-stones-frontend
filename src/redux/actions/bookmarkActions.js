import {
  BOOKMARK_REQUEST,
  BOOKMARK_SUCCESS,
  BOOKMARK_ERROR,
} from './actionTypes';

export const bookmarkRequest = slug => ({
  type: BOOKMARK_REQUEST,
  slug,
});

export const bookmarkSuccess = data => ({
  type: BOOKMARK_SUCCESS,
  data,
});

export const bookmarkError = error => ({
  type: BOOKMARK_ERROR,
  error,
});
