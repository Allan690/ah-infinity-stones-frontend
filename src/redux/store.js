/* eslint no-underscore-dangle: 0 */
/* eslint-disable no-undef */
import { applyMiddleware, createStore, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import allReducers from './reducers';
import rootSaga from './sagas';

// only activate redux devtools on the browser otherwise just compose
const composer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const composeEnhancers = composer;

const middleware = createSagaMiddleware();
const store = createStore(
  allReducers,
  composeEnhancers(applyMiddleware(middleware)),
);

export default store;

middleware.run(rootSaga);
