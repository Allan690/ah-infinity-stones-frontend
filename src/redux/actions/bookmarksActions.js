import { REQUEST_BOOKMARKS, BOOKMARKS_SUCCESS, BOOKMARKS_FAILURE } from './actionTypes';

export const requestBookmarks = () => ({
  type: REQUEST_BOOKMARKS,
});

export const bookmarksSuccess = data => ({
  type: BOOKMARKS_SUCCESS,
  data,
});

export const bookmarksFailure = error => ({
  type: BOOKMARKS_FAILURE,
  error,
});
