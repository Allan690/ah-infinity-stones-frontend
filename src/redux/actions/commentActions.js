import { GET_COMMENTS, POST_COMMENT, GET_TOPIC } from './actionTypes';

export const getComments = () => ({
  type: GET_COMMENTS,
});

export const postComment = postData => ({
  type: POST_COMMENT,
  payload: postData,
});

export const getTopic = () => ({
  type: GET_TOPIC,
});
