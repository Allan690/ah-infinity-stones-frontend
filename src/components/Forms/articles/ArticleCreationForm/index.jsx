import React, { Component } from 'react';
import { connect } from 'react-redux';
import './articleCreation.scss';
import 'react-draft-wysiwyg/dist/react-draft-wysiwyg.css';
import EditorToolbar from '../EditorToolbar';
import { createArticleAction } from '../../../../redux/actions/articleActions';


class ArticleCreation extends Component {
  constructor(props) {
    super(props);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.state = {
      articleTitle: '',
      articleDescription: '',
      articleBody: '',
      articleTags: '',
    };
    this.onChange = this.onChange.bind(this);
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  handleSubmit(event) {
    event.preventDefault();
    const article = {
      title: this.state.articleTitle,
      description: this.state.articleDescription,
      body: document.getElementById('articleBody').value,
      tag: this.state.articleTags.split(','),
    };
    this.props.createArticleAction(article);
  }

  render() {
    return (
      <div>
        <div className="container">
          <form id="articleForm" method="post" encType="multipart/form-data" onSubmit= {this.handleSubmit}>
            <ul className="article-creation-form">
              <li><label htmlFor="articleTitle">Title</label></li>
              <li><input type="text" name="articleTitle" id="articleTitle" placeholder="Title" required onChange={this.onChange} /></li>
              <li><label htmlFor="articleDescription">Description</label></li>
              <li><textarea name="articleDescription" id="articleDescription" placeholder="Add a two-line version of your story." required onChange={this.onChange} /></li>
              <li><label htmlFor="articleBody">Body</label></li>
              <div id="toolbar-container" />
              <li><EditorToolbar /></li>
              <li><label htmlFor="articleTags">Tags</label></li>
              <li><input type="text" name="articleTags" id="articleTags" placeholder="Tags" required onChange={this.onChange} /></li>
              <li>
                <button type="submit" className="btn btn-primary" id="publishBtn">
                  <i className="mdi mdi-pencil icon-publish" />
                  Publish
                </button>
              </li>
            </ul>
          </form>
        </div>
      </div>
    );
  }
}
const mapStateToProps = (state) => {
  return {
    article: state.articleReducer.article,
  };
};

const mapDispatchToProps = { createArticleAction };
export default connect(mapStateToProps, mapDispatchToProps)(ArticleCreation);
