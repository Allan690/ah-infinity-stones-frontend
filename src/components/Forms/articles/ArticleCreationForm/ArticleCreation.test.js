/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
/* eslint-disable no-shadow */
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import fetchMock from 'fetch-mock';
import createSagaMiddleware from 'redux-saga';
import { applyMiddleware, createStore } from 'redux';
import allReducers from '../../../../redux/reducers/index';
import store from '../../../../redux/store';
import ArticleCreationForm from './index';
import * as actions from '../../../../redux/actions/articleActions';
import { CREATE_ARTICLE, CREATE_ARTICLE_SUCCESS, CREATE_ARTICLE_FAILURE } from '../../../../redux/actions/actionTypes';


describe('async actions', () => {
  beforeEach(() => {
    const middleware = createSagaMiddleware();
    const store = createStore(
      allReducers, applyMiddleware(middleware),
    );
    const article = '{body: "You have to pray and play", description: "Ever wonder how?", title: "winning the jackpot", tags: "sportpesa"}';
    const expectedAction = {
      type: CREATE_ARTICLE,
      payload: article,
    };
  });
  afterEach(() => {
    fetchMock.restore();
  });
  it('creates CREATE_ARTICLE_SUCCESS when fetching articles has been done', () => {
    const article = '{body: "You have to pray and play", description: "Ever wonder how?", title: "winning the jackpot", tags: "sportpesa"}';
    const articleSuccess = '{art_slug: "winning-the-jackpot", author: "Ronny",body: "You have to pray and play", description: "Ever wonder how?"}';
    const error = '{type: "cors", url: "https://cors-anywhere.herokuapp.com/ah-infinites-staging.herokuapp.com/api/articles", redirected: false, status: 500, ok: false';

    fetchMock.getOnce('/articles', {
      body: article,
      headers: {
        Accept: 'application/json',
        'Content-type': 'application/json',
        Authorization: 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJlbWFpbCI6InJvbm55bWFnZWhAZ21haWwuY29tIiwidXNlcm5hbWUiOiJSb25ueSIsImV4cCI6MTU1MDQzMTkwM30.BFFFIlWsNj6mpKijYeDqi7rS-uK39Z4tLTXcivZyHHI',
      },
    });
    const expectedActions = [
      {
        type: CREATE_ARTICLE,
        payload: article,
      },
      {
        type: CREATE_ARTICLE_SUCCESS,
        payload: articleSuccess,
      },
      {
        type: CREATE_ARTICLE_FAILURE,
        payload: error,
      },
    ];
    const mockStore = configureMockStore();
    const myStore = mockStore({ article: [] });

    const res = myStore.dispatch(actions.createArticleAction(article));
    const res2 = myStore.dispatch(actions.createArticleSuccessAction(articleSuccess));
    expect(res).toEqual(expectedActions[0]);
    expect(res2).toEqual(expectedActions[1]);
  });
});
describe('Article Creation Component', () => {
  let articleCreationContainer;
  let reduxArticleCreationForm;
  beforeEach(() => {
    articleCreationContainer = shallow(<ArticleCreationForm />);
    reduxArticleCreationForm = mount(
      <Provider store={store}>
        <BrowserRouter>
          <ArticleCreationForm />
        </BrowserRouter>
      </Provider>,
    );
  });
  it('ArticleCreationForm renders successfully', () => {
    expect(articleCreationContainer.exists()).toEqual(true);
  });
  it('render 1 Article Creation component', () => {
    expect(articleCreationContainer).toHaveLength(1);
  });
  it('article creation component should match snapshot', () => {
    expect(articleCreationContainer.getElements()).toMatchSnapshot();
  });
});

describe('Action Creators', () => {
  let articleCreationContainer;
  let reduxArticleCreationForm;
  beforeEach(() => {
    articleCreationContainer = shallow(<ArticleCreationForm />);
    reduxArticleCreationForm = mount(
      <Provider store={store}>
        <BrowserRouter>
          <ArticleCreationForm />
        </BrowserRouter>
      </Provider>,
    );
  });
  it('should send a POST request all', () => {
    const article = '{body: "You have to pray and play", description: "Ever wonder how?", title: "winning the jackpot", tags: "sportpesa"}';
    const expectedAction = {
      type: CREATE_ARTICLE,
      payload: article,
    };
    expect(actions.createArticleAction(article)).toEqual(expectedAction);
  });
  it('should update the article to the state if successful', () => {
    const article = '{art_slug: "how-to-make", author: "Ronny",body: "You have to pray", description: "Ever wonder how?"}';
    const expectedAction = {
      type: CREATE_ARTICLE_SUCCESS,
      payload: article,
    };
    expect(actions.createArticleSuccessAction(article)).toEqual(expectedAction);
  });
  it('should catch error', () => {
    const error = '{type: "cors", url: "https://cors-anywhere.herokuapp.com/ah-infinites-staging.herokuapp.com/api/articles", redirected: false, status: 500, ok: false';
    const expectedAction = {
      type: CREATE_ARTICLE_FAILURE,
      payload: error,
    };
    expect(actions.createArticleFailureAction(error)).toEqual(expectedAction);
  });
});
