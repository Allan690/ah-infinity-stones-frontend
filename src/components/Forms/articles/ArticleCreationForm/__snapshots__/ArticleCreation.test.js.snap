// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Creation Component article creation component should match snapshot 1`] = `
Array [
  <Context.Consumer>
    [Function]
  </Context.Consumer>,
]
`;
